lib/Support/Check.cpp

```diff
@@ -60,40 +82,54 @@ static bool testBufferExact(offloadtest::Buffer *B1, offloadtest::Buffer *B2) {
   return true;
 }
 
+template <typename T>
+static bool testAll(std::function<bool(const T &, const T &)> Fn,
+                    llvm::ArrayRef<T> Arr1, llvm::ArrayRef<T> Arr2) {
+  if (Arr1.size() != Arr2.size())
+    return false;
+
+  for (unsigned I = 0, E = Arr1.size(); I < E; ++I) {
```

Comment:
I think 'I' should be a size_t for consistency with the return type of llvm::ArrayRef.size()? Although I doubt it would ever cause a real issue.